src/Sarif.Multitool.FunctionalTests/SarifMultitoolTestBase.cs

```diff
@@ -46,38 +46,38 @@ private static string MakeQualifiedFilePath(string testDirectory, string testFil
 
         private static void SelectiveCompare(SarifLog actualLog, SarifLog expectedLog)
         {
-            Result[] actualResults = SafeListToArray(actualLog.Runs[0].Results);
```

Comment:
>SafeListToArray [](start = 37, length = 15)

I don't think you can remove this. The spec says that `Runs` can be `null` if (for example) the tool couldn't start, or the database query that was supposed to return a set of runs was malformed. Removing this from _test_ code relies on us never having a test with a missing `Runs` array.

Reply:
How about `actualLog.Runs?[0].Results` -- checking to see if that's legal C#.

---
In reply to: [249150331](https://github.com/Microsoft/sarif-sdk/pull/1223#discussion_r249150331) [](ancestors = 249150331)

Reply:
I'm sorry, I'm confused, but I still have a point :-). You don't need the helper, because you're not converting a possibly-`null` list to an array. BUT! `Runs` might still be `null`. I guess it's ok to leave. On the day we add such a test, that will be time enough to test for that.

---
In reply to: [249150739](https://github.com/Microsoft/sarif-sdk/pull/1223#discussion_r249150739) [](ancestors = 249150739,249150331)